### DISCUSSION TITLE
focus-within don't working when have hover variant

### DISCUSSION THREAD

**Discussion by zeing** [tailwindlabs/tailwindcss#5350] [Help] [answered]
```
<div className="flex space-x-4 text-12 w-full border-b border-gray-300 pb-3 focus-within:border-primary hover:border-gray-400 ">       
    <input
      className="w-full outline-none"
      placeholder="Search"
      onChange={(e) => onChange(e.target.value)}
    ></input>
</div>
```
or https://play.tailwindcss.com/NopVdgFRcu

When I hover or focus with the input I want it to have `border-b` in `div` but `focus-within` .it doesn't work because `hover` is still active

I have been tried to remove `hover:border-gray-400` then `focus-within:border-primary` works

How to fix it?

**Comment by simonswiss** [comment, marked as answer]
Hey!

You can chain the two variants here: `hover:focus-within:border-primary`.

That way, when you hover over an element which also happens to contained an element in focus, the correct color will be applied, and override the color you apply on "non focus-within" hover 👍

I have updated your Tailwind Play to demonstrate that:

https://play.tailwindcss.com/COT0Aw8bo3

---

**Bonnus**, if you use the [`Just In Time` mode](https://tailwindcss.com/docs/just-in-time-mode) (`mode: 'jit'` in your Tailwind config file), you can completely remove your `variants` object where you add variants like `focus-within`, they will all be enabled by default, and you can compose (chain) as many variants as you want in any order you want 🎉

![image](https://user-images.githubusercontent.com/485747/131945951-3e627100-a205-42c3-a221-090e22ddd0b9.png)

**Reply by zeing** [reply]
Oh Thank you so much 

May you have `Tailwind Play` for JIT mode ?

**Reply by zeing** [reply]
@simonswiss 
In tailwind.config.js why borderColor can\'t in extend  like

don\'t work

```
variants: {
    extend: {
     borderColor: [
      'active',
      'focus',
      'disabled',
      'responsive',
      'hover',
      'focus-within',
    ],
    },
  },
```

works
```
variants: {
     borderColor: [
      'active',
      'focus',
      'disabled',
      'responsive',
      'hover',
      'focus-within',
    ],
  },
```

**Reply by simonswiss** [reply]
Hey!

Tailwind Play already supports `JIT`, and when you start a new Play, the `mode: 'jit'` is in the config file by default.

And to answer your question - when you extend variants, here's the order in which they get respected (whichever comes later wins):

<img width="891" alt="image" src="https://user-images.githubusercontent.com/485747/131965702-da711685-917f-4874-b414-44b3ec2d0f79.png">

As you can see, `focus` comes _after_ `focus-within`, so that's why hover takes over in this scenario.

You can change that behaviour by customising the `variantOrder` array:

https://tailwindcss.com/docs/configuring-variants#ordering-custom-variants

This is one of the main things the `JIT` engine makes so much better, you can completely forget about all those variant extensions and use them in any order/stacking, so I recommend you play around with the `JIT` mode 👍

**Reply by zeing** [reply]
thank you so much

**Reply by simonswiss** [reply]
You're very welcome! ✨
